capsules/src/alarm.rs

```diff
@@ -197,8 +179,13 @@ impl<'a, A: Alarm> time::Client for AlarmDriver<'a, A> {
 
         // If there are armed alarms left, reset the underlying alarm to the
         // nearest interval.  Otherwise, disable the underlying alarm.
-        if self.num_armed.get() > 0 {
-            self.reset_active_alarm();
+        if self.num_armed.get() == 0 {
+            self.alarm.disable();
+        } else if let Some(next_alarm) = self.reset_active_alarm(now) {
```

Comment:
Should this if statement be a while loop instead? It feels like this code only checks the next alarm to see if it already fired, whereas there could be multiple.

Reply:
Potentially, yeah... Fix now or in a follow up PR? I don't believe this would introduce _new_ regressions, but I _think_ you're right that this could result in missing a second expired alarm.

Reply:
A follow up PR seems sufficient. This is pretty darn unlikely to occur in practice